### PR TITLE
endpoint to get external cluster upgrades

### LINKED ISSUE #8109: Add endpoint to get external cluster upgrades
Add endpoint to get external cluster upgrades

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: endpoint to get external cluster upgrades

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8109

```release-note
endpoint to get external cluster upgrades 
GET /api/v2/projects/{project_id}/kubernetes/clusters/{cluster_id}/upgrades
```

### CODE REVIEW

File: pkg/handler/v2/external_cluster/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+		if cloud.GKE != nil {
+			sa, err := secretKeySelector(cloud.GKE.CredentialsReference, resources.GCPServiceAccount)
+			if err != nil {
+				return nil, err
+			}
+			return providercommon.ListGKEUpgrades(ctx, sa, cloud.GKE.Zone)
+		}
+
+		return nil, fmt.Errorf("wrong cloud structure")
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/upgrade.go]: NIt: can we have a better error message here

File: pkg/handler/v2/external_cluster/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+		cluster, err := getCluster(ctx, userInfoGetter, clusterProvider, privilegedClusterProvider, project.Name, req.ClusterID)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		upgrades := make([]*apiv1.MasterVersion, 0)
+		cloud := cluster.Spec.CloudSpec
+		if cloud == nil {
+			return upgrades, nil
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/upgrade.go:58]: Nit: should we return an error message or just a empty list here

Comment (zreigz) [reply pkg/handler/v2/external_cluster/upgrade.go:58]: for read-only cluster we can return an empty list which means nothing to upgrade

File: pkg/handler/v2/external_cluster/upgrade.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/v2/external_cluster/upgrade.go]: done

**Review by imharshita — APPROVED** [review]

**Pull request merged**